llama_index/llms/openai.py

```diff
@@ -68,10 +68,16 @@ class OpenAI(LLM):
     )
     max_retries: int = Field(description="The maximum number of API retries.")
 
-    api_key: str = Field(default=None, description="The OpenAI API key.", exclude=True)
```

Comment:
Having `default=None` was a type mismatch, even though it gets updated in `__init__`